contracts/modules/governance/MemberRoles.sol

```diff
@@ -207,24 +207,33 @@ contract MemberRoles is IMemberRoles, Governed, MasterAwareV2 {
   }
 
   /// Withdraws membership
-  function withdrawMembership() public {
+  function withdrawMembership() public whenNotPaused {
 
     ITokenController _tokenController = tokenController();
-    require(!master.isPause() && isMember(msg.sender));
-    // No locked tokens for Member/Governance voting
-    require(block.timestamp > token.isLockedForMV(msg.sender));
+
+    require(isMember(msg.sender), "MemberRoles: Caller is not a member");
+
+    require(
+      block.timestamp > token.isLockedForMV(msg.sender),
+      "MemberRoles: Locked for voting in governance"
+    );
+
+    require(
+      tokenController().isStakingPoolManager(msg.sender) == false,
```

Comment:
say you have a staking pool, you want to withdraw your address from membership.

The flow is to prepare another address, send an offer for staking pool ownership and wait for it to be accepted.

In an ideal case i'm thinking i'd want to be able to just exit the system if my staking pools have no NXM stake, without having to go though this ownership transfer flow (which involves another member or me with another member address to accept a potentially empty staking pool).

This is somewhat of an edge case since its termination, i wanted to walk through it to see if we are all on-board with how it goes.

Reply:
valid point. unsure if there is an easy way to do so, but regardless i suggest creating an issue for this and treating it after the launch

Reply:
Agreed let's make the issue.